crates/zk-por-core/src/bin/prover.rs

```diff
@@ -0,0 +1,154 @@
+use itertools::Itertools;
+use plonky2::{hash::hash_types::HashOut, plonk::proof::ProofWithPublicInputs};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::{env, fs::File, io::Write, path::PathBuf, str::FromStr};
+use zk_por_core::{
+    account::Account,
+    circuit_config::{STANDARD_CONFIG, STANDARD_ZK_CONFIG},
+    circuit_registry::registry::CircuitRegistry,
+    config::ProverConfig,
+    e2e::{batch_prove_accounts, recursive_prove_subproofs},
+    parser::{self, AccountParser, FilesCfg, FilesParser},
+    types::{C, D, F},
+};
+use zk_por_tracing::{init_tracing, TraceConfig};
+
+#[derive(Serialize, Deserialize)]
+struct Proof {
+    round_num: usize,
+    root_vd_digest: HashOut<F>,
+    proof: ProofWithPublicInputs<F, C, D>,
+}
+
+fn main() {
+    let cfg = ProverConfig::try_new().unwrap();
+    let trace_cfg: TraceConfig = cfg.log.into();
+    init_tracing(trace_cfg);
+
+    const RECURSION_FACTOR: usize = 64;
+    const BATCH_PROVING_THREADS_NUM: usize = 4;
+    const RECURSIVE_PROVING_THREADS_NUM: usize = 2;
+
+    if cfg.prover.hyper_tree_size as usize != RECURSION_FACTOR {
+        panic!("The hyper_tree_size is not configured to be equal to 64 (Recursion_Factor)");
+    }
+    let batch_size = cfg.prover.batch_size as usize;
+    let asset_num = cfg.prover.num_of_tokens as usize;
+
+    // the path to dump the final generated proof
+    let mut bench_mode = true;
+    let args: Vec<String> = env::args().collect();
+    let arg1 = args.get(1).expect(
```

Comment:
as in pay 2 commitment. we use cli with --help. just for better user experience; we can change later